core/test/org/javarosa/xpath/expr/test/XPathPathExprTest.java

```diff
@@ -0,0 +1,171 @@
+import org.javarosa.core.model.condition.IFunctionHandler;
+import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.core.model.data.IntegerData;
+import org.javarosa.core.model.data.StringData;
+import org.javarosa.core.model.instance.FormInstance;
+import org.javarosa.core.model.instance.TreeElement;
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.core.model.utils.DateUtils;
+import org.javarosa.model.xform.XPathReference;
+import org.javarosa.xpath.IExprDataType;
+import org.javarosa.xpath.XPathException;
+import org.javarosa.xpath.XPathParseTool;
+import org.javarosa.xpath.XPathTypeMismatchException;
+import org.javarosa.xpath.XPathUnhandledException;
+import org.javarosa.xpath.XPathUnsupportedException;
+import org.javarosa.xpath.expr.XPathExpression;
+import org.javarosa.xpath.expr.XPathFuncExpr;
+import org.javarosa.xpath.expr.XPathNumericLiteral;
+import org.javarosa.xpath.expr.XPathPathExpr;
+import org.javarosa.xpath.parser.XPathSyntaxException;
+import org.javarosa.xml.TreeElementParser;
+import org.javarosa.xml.util.InvalidStructureException;
+
+import org.kxml2.io.KXmlParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+/**
+ *
+ * @author Phillip Mates
+ */
+
+public class XPathPathExprTest extends TestCase {
+
+    private static final String formPath = new String("/test_xpathpathexpr.xml");
+    // private static final String formPath = new String("/test_small_xml_example.xml");
+
+    public XPathPathExprTest(String name, TestMethod rTestMethod) {
+        super(name, rTestMethod);
+    }
+
+    public XPathPathExprTest(String name) {
+        super(name);
+    }
+
+    public XPathPathExprTest() {
+        super();
+    }
+
+    public Test suite() {
+        TestSuite aSuite = new TestSuite();
+
+        aSuite.addTest(new XPathPathExprTest("XPath Path Expression Test", new TestMethod() {
+            public void run(TestCase tc) {
+                ((XPathPathExprTest) tc).doTests();
+            }
+        }));
+
+        return aSuite;
+    }
+
+    public void doTests() {
+        KXmlParser parser = null;
+        TreeElement root = null;
+        EvaluationContext ec = new EvaluationContext(null);
+
+        // read in xml
+        try {
+            InputStream is = System.class.getResourceAsStream(formPath);
+            parser = new KXmlParser();
+            parser.setInput(is, "UTF-8");
+        } catch (XmlPullParserException e) {
+            fail("Contents at filepath could not be parsed as XML: " + formPath);
+            return;
+        }
+
+        // turn parsed xml into a form instance
+        try {
+            root = new TreeElementParser(parser, 0, "data").parse();
+        } catch (Exception e) {
+            fail("File couldn't be parsed into a TreeElement: " + formPath);
+            return;
+        }
+        FormInstance instance = null;
+        try {
+            instance = new FormInstance(root.getChildAt(0), "data");
```

Comment:
root.getChildAt(0) is necessary, otherwise FormInstance crashes. Maybe I'm parsing things incorrectly...